### PR TITLE
common/resolver: registries use global registrar

### PR DESCRIPTION
**Pull request by zelig**
* global registrar now in resolver, console jsre refers to resolver.abi/addr, geth/contracts.go removed
* add Call to resolver backend interface
* global registrar name registry interface in resolver
* the hashReg and UrlHing contracts now initialised from global registry
* if not registered one can call the CreateContracts which creates them, reserved and registers them
* tests
* newRegistry returns the contract addresses for HashReg and UrlHint

### CODE REVIEW

File: cmd/geth/admin.go [hunk 1]
```diff
@@ -781,22 +781,22 @@ func (js *jsre) newRegistry(call otto.FunctionCall) otto.Value {
...
-	err = registry.CreateContracts(common.HexToAddress(addr))
+	hashReg, urlHint, err = registry.CreateContracts(common.HexToAddress(addr))
 	if err != nil {
 		fmt.Println(err)
-		return otto.FalseValue()
+		return otto.UndefinedValue()
 	}
 
-	return otto.TrueValue()
+	return js.re.ToVal([]string{hashReg, urlHint})
```
Comment (fjl) [inline cmd/geth/admin.go]: Please update this to use `call.Otto.ToValue`.

Comment (zelig) [inline cmd/geth/admin.go]: thanks. done.

**Comment by zelig** [comment]
rebased and ready

**Comment by fjl** [comment]
Sorry for taking so long, will try it and merge tonight.

**Comment by zelig** [comment]
thanks @fjl , but actually leave it for now, I will be adding a few changes and add them. Put back in progress

**Comment by zelig** [comment]
superseded by #1322

**Pull request closed without merging**